app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,77 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'tweets/create_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content].empty?
+      redirect to "/tweets/new"
+    else
+      tweet = current_user.tweets.create(content: params[:content])
+      redirect to "/tweets/#{tweet.id}"
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
```

Comment:
It would be best to verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit.